src/box/memtx_hash.cc

```diff
@@ -371,16 +371,14 @@ memtx_hash_index_replace(struct index *base, struct tuple *old_tuple,
 		if (pos == light_index_end)
 			pos = light_index_insert(hash_table, h, new_tuple);
 
-		ERROR_INJECT(ERRINJ_INDEX_ALLOC,
-		{
+		ERROR_INJECT(ERRINJ_HASH_INDEX_REPLACE, {
 			light_index_delete(hash_table, pos);
 			pos = light_index_end;
 		});
 
 		if (pos == light_index_end) {
-			diag_set(OutOfMemory,
-				 (ssize_t)light_index_count(hash_table),
-				 "hash_table", "key");
+			diag_set(ClientError, ER_HASH_INDEX_REPLACE_FAILED,
+				 (unsigned)light_index_count(hash_table));
```

Comment:
This doesn't necessarily happen because of the matras limit. It also happens when we reach the memtx arena limit. So I think that the error type should remain `OutOfMemory`. Also, changing the error type would be inconsistent with the BPS tree implementation where we [raise](https://github.com/tarantool/tarantool/blob/0ea1ba877b8e62e0acb558b55623d93bf58a0e1f/src/box/memtx_tree.cc#L1135) `OutOfMemory`.

I don't see much point in including the index size into the error message because the user can check it out with `index.count` when they investigate. Anyway, if we decide to add it, it's better to include this information as a [payload](https://github.com/tarantool/tarantool/blob/0ea1ba877b8e62e0acb558b55623d93bf58a0e1f/src/lib/core/diag.h#L146), and this should be done consistently for all memtx indexes.

Reply:
OK, but what number of bytes to return with `OutOfMemory` here?

Reply:
`MEMTX_EXTENT_SIZE`, like in the tree index.

Reply:
Done.